ladybug_geometry/polyskel.py

```diff
@@ -0,0 +1,558 @@
+# coding=utf-8
+"""
+Implementation of the straight skeleton algorithm as described by Felkel and Obdržálek in their 1998 conference paper Straight skeleton implementation.
+"""
+
+from __future__ import division
+
+import logging
+import heapq
+from itertools import *
+from collections import namedtuple
+import operator
+
+# Geometry classes
+from ladybug_geometry.geometry2d.polygon import Polygon2D
+from ladybug_geometry.geometry2d.pointvector import Point2D, Vector2D
+from ladybug_geometry.geometry2d.line import LineSegment2D
+from ladybug_geometry.geometry2d.ray import Ray2D
+from ladybug_geometry import intersection2d
+
+log = logging.getLogger("__name__")
+
+
+# For Debugging
+
+class Debug:
+	def __init__(self, image):
+		if image is not None:
+			self.im = image[0]
+			self.draw = image[1]
+			self.do = True
+		else:
+			self.do = False
+
+	def line(self, *args, **kwargs):
+		if self.do:
+			self.draw.line(*args, **kwargs)
+
+	def rectangle(self, *args, **kwargs):
+		if self.do:
+			self.draw.rectangle(*args, **kwargs)
+
+	def show(self):
+		if self.do:
+			self.im.show()
+
+_debug = Debug(None)
+
+def set_debug(image):
+	global _debug
+	_debug = Debug(image)
+
+
+# Skeleton Code
+
+def _window(lst):
+	prevs, items, nexts = tee(lst, 3)
+	prevs = islice(cycle(prevs), len(lst)-1, None)
+	nexts = islice(cycle(nexts), 1, None)
+	return zip(prevs, items, nexts)
+
+def _determinant(a, b):
+	res = a.x*b.y - b.x*a.y
```

Comment:
You need space before and after *. We have to configure PEP8 on your editor.

```suggestion
	res = a.x * b.y - b.x * a.y
```

Reply:
Yeah, these aren't highlighted as PEP8 violations in my IDE. I'll have to search for a better plugin.